lib/test/hip/RunGEMMKernelTileSelection_test.cpp

```diff
@@ -0,0 +1,253 @@
+        HIP_CHECK_EXC(hipMemcpy(b_d,     b_h.data(),     problem.b().totalAllocatedBytes(), hipMemcpyHostToDevice));
+        HIP_CHECK_EXC(hipMemcpy(c_d,     c_h.data(),     problem.c().totalAllocatedBytes(), hipMemcpyHostToDevice));
+        HIP_CHECK_EXC(hipMemcpy(d_d,     d_in_h.data(),  problem.d().totalAllocatedBytes(), hipMemcpyHostToDevice));
+        HIP_CHECK_EXC(hipMemcpy(d_ref_d, d_ref_h.data(), problem.d().totalAllocatedBytes(), hipMemcpyHostToDevice));
+
+        inputs.a = a_d;
+        inputs.b = b_d;
+        inputs.c = c_d;
+        inputs.d = d_d;
+
+        inputs.alpha = RandomInt<float>(rng)();
+        if(problem.beta() == 1.0)
+            inputs.beta = 1.0;
+        else if(problem.beta() == 0.0)
+            inputs.beta = 0.0;
+        else
+            inputs.beta = RandomInt<float>(rng)();
+
+        hardware = hip::GetCurrentDevice();
+        ASSERT_NE(hardware, nullptr);
+
+#if 1
+        TypedContractionInputs<float> inputsRefHost;
+        inputsRefHost.a = a_h.data();
+        inputsRefHost.b = b_h.data();
+        inputsRefHost.c = c_h.data();
+        inputsRefHost.d = d_ref_h.data();
+        inputsRefHost.alpha = inputs.alpha;
+        inputsRefHost.beta = inputs.beta;
+
+
+        Client::SolveCPU(problem, inputsRefHost);
+#else
+        rocblas_handle roc = nullptr;
+        ASSERT_RB(rocblas_create_handle(&roc));
+
+        for(int i = 0; i < problem.batchSize(0); i++)
+        {
+            size_t a_offset = problem.a().index(0,0,i);
+            size_t b_offset = problem.b().index(0,0,i);
+            size_t d_offset = problem.d().index(0,0,i);
+
+            auto transA = problem.transA() ? rocblas_operation_transpose : rocblas_operation_none;
+            auto transB = problem.transB() ? rocblas_operation_transpose : rocblas_operation_none;
+
+            ASSERT_RB(rocblas_sgemm(roc, transA, transB,
+                                    problem.freeSizeA(0), problem.freeSizeB(0), problem.boundSize(0),
+                                    &inputs.alpha, a_d + a_offset, problem.a().strides()[1],
+                                    b_d + b_offset, problem.b().strides()[1],
+                                    &inputs.beta, d_ref_d + d_offset, problem.d().strides()[1]));
+        }
+
+        HIP_CHECK_EXC(hipMemcpy(d_ref_h.data(), d_ref_d, problem.d().totalAllocatedBytes(), hipMemcpyDeviceToHost));
+
+        ASSERT_RB(rocblas_destroy_handle(roc));
+#endif
+
+	}
+	
+    void TearDown() override
+    {
+        hipFree(a_d);
+        hipFree(b_d);
+        hipFree(c_d);
+        hipFree(d_d);
+        hipFree(d_ref_d);
+
+        hipDeviceReset();
+    }
+};
+
+TEST_P(RunGEMMKernelSolutionSelectionTest , KernelsTileSelection)
+{
+
+    ContractionProblem problem = GetParam();
+    //std::cout << problem << std::endl;
+
+    auto library = LoadLibraryFile<ContractionProblem>(TestData::File("kernels_tile_selection/TensileLibrary.yaml").native());
```

Comment:
Is this built manually?

Reply:
Currently this is built manually, Until I get the python part that generates it for the logic and solution data.

Reply:
I don't have this tests enabled so (it is not included in CMakeList.txt).

I don't have the python implementation which will generate the related solution. That will be part of my next pull request. When that is done I will enable the test.